fix: code quality issues

withshubh opened the pull request:
### Description
Hi :wave: I work at [DeepSource](https://deepsource.io), I ran DeepSource analysis on the forked copy of this repo and found some interesting [code quality issues](https://deepsource.io/gh/withshubh/oppia/issues/?category=recommended) in the codebase, opening this PR so you can assess if our platform is right and helpful for you.


### Motivation and Context
few fixes boost minor performance and few removes the anti-patterns in code.

### Summary of changes

- Removed unnecessary use of comprehension
- Removed unnecessary comprehension
- Removed redundant call
- Refactored `if` expression
- Added .deepsource.toml to continuously analyze and detect code quality issues

## Essential Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

withshubh committed:
Add .deepsource.toml

Signed-off-by: shubhendra <withshubh@gmail.com>

withshubh committed:
Remove unnecessary use of comprehension

Signed-off-by: shubhendra <withshubh@gmail.com>

withshubh committed:
Remove redundant call

Signed-off-by: shubhendra <withshubh@gmail.com>

withshubh committed:
Remove unnecessary comprehension

Signed-off-by: shubhendra <withshubh@gmail.com>

withshubh committed:
Refactor `if` expression

Signed-off-by: shubhendra <withshubh@gmail.com>

nithusha21 reviewed (changes requested):
Hey, could you give us more context about why these changes were made? It seems to violate our lint checks which isn't good. And I am a bit skeptical about adding a new config file

withshubh commented:
> Hey, could you give us more context about why these changes were made? It seems to violate our lint checks which isn't good. And I am a bit skeptical about adding a new config file

Hey @nithusha21 :wave: 

- Removed unnecessary use of comprehension
> boosts minor performance, know more [here](https://deepsource.io/gh/withshubh/oppia/issue/PYL-R1721/description/)
- Removed unnecessary comprehension 
> boosts minor performance, know more [here](https://deepsource.io/gh/withshubh/oppia/issue/PTC-W0016/description/)
- Removed redundant call
> fixes antipatter, know more [here](https://deepsource.io/gh/withshubh/oppia/issue/PTC-W0017/description)
- Refactored if expression
> fixes antipattern, know more [here](https://deepsource.io/gh/withshubh/oppia/issue/PYL-R1719/description)

Adding this configuration will let you activate the continuous analysis of the repo to fix code quality issues and automate code reviews. You don't have to add [DeepSource](https://deepsource.io/) to the CI, it is a service that runs parallel to CI. You can use its auto-fix feature to fix most of the issues in the codebase.

sajalasati commented:
@nithusha21 @sagangwee PTAL at the new changes being introduced by this PR!
@withshubh Did you skip the pre-push checks while making this PR? Please ensure that at least the lint checks and backend tests pass on CI if you want the refactored changes to be merged. Also make sure you go through our contributor guidelines in [Wiki](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia), and follow the right steps for making a PR and addressing review comments. Thanks!
